portal/server/routes/admin/activity.ts

```diff
@@ -0,0 +1,94 @@
+import { Request, Router } from 'express';
+import { validationErrorHandler } from '../../helpers';
+
+const router = Router();
+
+type ActivitySearchPostRequestBody = {
+  usersSearchTerm?: string;
+}
+
+router.get('/activity/search', (req, res) => res.render('admin/activity/search-users.njk', { user: req.session.user, primaryNav: 'activity' }));
+
+router.post('/activity/search', (req: Request<unknown, unknown, ActivitySearchPostRequestBody>, res) => {
+  const { usersSearchTerm } = req.body;
+
+  if (!usersSearchTerm) {
+    return res.render('admin/activity/search-users.njk', {
+      user: req.session.user,
+      errors: validationErrorHandler([
+        {
+          errMsg: 'Enter the name or email address of the user you wish to be featured in the report',
+          errRef: 'usersSearchTerm',
+        },
+      ]),
+      primaryNav: 'activity',
+    });
+  }
+  if (usersSearchTerm.length < 3) {
+    return res.render('admin/activity/search-users.njk', {
+      user: req.session.user,
+      errors: validationErrorHandler([
+        {
+          errMsg: 'The name or email address of the user must contain at least 3 characters',
+          errRef: 'usersSearchTerm',
+        },
+      ]),
+      primaryNav: 'activity',
+    });
+  }
+
+  return res.redirect(301, `/admin/activity/search-results?q=${encodeURIComponent(usersSearchTerm)}`);
+});
+
+router.get('/activity/search-results', (req, res) => {
+  const searchTerm = req.query.q as string | undefined;
+
+  if (!searchTerm || searchTerm.length < 3) {
```

Comment:
Unsure if I need to add error handling here as the post request deals with this. It could be accessed by typing in the URL manually